File: 1-Beginner/sieve_of_Eratosthenes/python/cmd.py
def obtaining_prime_list(n):
    
    prime = [True for i in range(n+1)]
    p = 2 #the first prime number
    sum = 0
    while (p*p <= n ):
        
        # If prime[p] is not changed, then it is a prime xD
        if (prime[p]==True):
            #updating the multiples of p
            for i in range(p*p,n+1,p):
                prime[i]= False
        p += 1

        for p in range(2,n):
            if prime[p]:
                print(p)
                sum += p
    return sum
print("10",obtaining_prime_list(10))
#print("2mil",obtaining_prime_list(2000000))
#print("sec",obtaining_prime_list(2000001))
#print("asdas",obtaining_prime_list(2000002))

File: 1-Beginner/sieve_of_Eratosthenes/C/cmd.c
#include <stdio.h>


int main()
{

  int n = 10;
  int prime[n];
  int sum = 0;
  // creating an array with True Values
  for (int i =0; i < n+1;i++)
  {
    prime[i] = 0;
  }

  for (int p = 2; p *p <=n;p++)
  {
    if (prime[p] == 0)
    {
      for (int i = p*p;i<=n;i+=p)
      {
        prime[i] = p;
      }
    }
  }

  for (int p = 2;p <= n;p++)
  {
    if (prime[p]== 0)
    {
      printf("%d\n",p);
      sum += p;
    }
  }


  printf("sum = %d\n",sum);
  return 0;
}
